core/src/cluster_info.rs

```diff
@@ -1731,19 +1737,21 @@ impl ClusterInfo {
         Ok(())
     }
 
-    fn handle_adopt_shred_version(self: &Arc<Self>, adopt_shred_version: &mut bool) {
-        // Adopt the entrypoint's `shred_version` if ours is unset
-        if *adopt_shred_version {
-            // If gossip was given an entrypoint, look up the ContactInfo by the given
-            // entrypoint gossip adddress
-            let gossip_addr = self.entrypoint.read().unwrap().as_ref().map(|e| e.gossip);
+    fn process_entrypoint(self: &Arc<Self>, entrypoint_processed: &mut bool) {
```

Comment:
handle_adopt_shred_version -> process_entrypoint, because this function now updates the entrypoint's id even when not shred adopting.

This function will get slightly more involved when multiple entrypoints are supported, when it'll adopt the shred of the first entrypoint found and then update the id for all entrypoints as they are discovered.